test/functional/system/rounds.js

```diff
@@ -0,0 +1,1538 @@
+					height,
+				})
+				.then(rows => {
+					// Normalize blocks
+					return library.modules.blocks.utils.readDbRows(rows);
+				});
+		},
+		getBlocks: round => {
+			return library.db.query(
+				'SELECT * FROM blocks WHERE CEIL(height / 101::float)::int = ${round} ORDER BY height ASC',
+				{ round }
+			);
+		},
+		getRoundRewards: round => {
+			return library.db
+				.query(
+					'SELECT ENCODE("publicKey", \'hex\') AS "publicKey", SUM(fees) AS fees, SUM(reward) AS rewards FROM rounds_rewards WHERE round = ${round} GROUP BY "publicKey"',
+					{ round }
+				)
+				.then(rows => {
+					const rewards = {};
+					_.each(rows, row => {
+						rewards[row.publicKey] = {
+							publicKey: row.publicKey,
+							fees: row.fees,
+							rewards: row.rewards,
+						};
+					});
+					return rewards;
+				});
+		},
+		getVoters: () => {
+			return library.db.query(
+				'SELECT "dependentId", ARRAY_AGG("accountId") FROM mem_accounts2delegates GROUP BY "dependentId"'
+			);
+		},
+	};
+
+	// Set rewards start at 150-th block
+	constants.rewards.offset = 150;
+
+	localCommon.beforeBlock('lisk_functional_rounds', lib => {
+		library = lib;
+
+		validateAccountsBalancesQuery = new QueryFile(
+			path.join(
+				__dirname,
+				'../common/sql/rounds/validate_accounts_balances.sql'
+			),
+			{ minify: true }
+		);
+
+		generateDelegateListPromise = Promise.promisify(
+			library.modules.delegates.generateDelegateList
+		);
+
+		addTransactionsAndForgePromise = Promise.promisify(
+			localCommon.addTransactionsAndForge
+		);
+	});
+
+	function getMemAccounts() {
+		return Queries.getAccounts().then(rows => {
+			const accounts = {};
+			_.map(rows, acc => {
+				accounts[acc.address] = acc;
+			});
+			return _.cloneDeep(accounts);
+		});
+	}
+
+	function getDelegates() {
+		return Queries.getDelegates().then(rows => {
+			const delegates = {};
+			_.map(rows, d => {
+				d.publicKey = d.publicKey.toString('hex');
+				delegates[d.publicKey] = d;
+			});
+			return _.cloneDeep(delegates);
+		});
+	}
+
+	function expectedMemState(transactions, _accounts) {
+		const accounts = _.cloneDeep(_accounts);
+		const lastBlock = library.modules.blocks.lastBlock.get();
+
+		// Update last block forger account
+		const found = _.find(accounts, {
+			publicKey: Buffer.from(lastBlock.generatorPublicKey, 'hex'),
+		});
+		if (found) {
+			found.producedBlocks += 1;
+			found.blockId = lastBlock.id;
+		}
+
+		// Mutate states - apply every transaction to expected states
+		_.each(transactions, transaction => {
+			// SENDER: Get address from senderId or if not available - get from senderPublicKey
+			let address =
+				transaction.senderId ||
+				library.modules.accounts.generateAddressByPublicKey(
+					transaction.senderPublicKey
+				);
+
+			// If account with address exists - set expected values
+			if (accounts[address]) {
+				// Update sender
+				accounts[address].balance = new Bignum(accounts[address].balance)
```

Comment:
we don't need to minus the transaction amount?

Reply:
We need and we do. :)